src/Guesser/TypeGuesser.php

```diff
@@ -44,31 +45,88 @@ public function guessType($class, $property, ModelManagerInterface $modelManager
         }
 
         switch ($metadata->getTypeOfField($propertyName)) {
-            case 'collection':
-            case 'hash':
+            case Type::COLLECTION:
+            case Type::HASH:
+                return new TypeGuess('array', [], Guess::HIGH_CONFIDENCE);
             case 'array':
-              return new TypeGuess('array', [], Guess::HIGH_CONFIDENCE);
-            case 'boolean':
+                @trigger_error(
+                    'The array type is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
+                    E_USER_DEPRECATED
+                );
+
+                return new TypeGuess('array', [], Guess::HIGH_CONFIDENCE);
+            case Type::BOOL:
+            case Type::BOOLEAN:
                 return new TypeGuess('boolean', [], Guess::HIGH_CONFIDENCE);
             case 'datetime':
+                @trigger_error(
+                    'The array datetime is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
+                    E_USER_DEPRECATED
+                );
+
+                return new TypeGuess('datetime', [], Guess::HIGH_CONFIDENCE);
             case 'vardatetime':
+                @trigger_error(
+                    'The array vardatetime is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
```

Comment:
```suggestion
                    'The vardatetime type is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x, to be removed in 4.0.'.
```